datastore/datastore.go

```diff
@@ -37,23 +50,35 @@ type KV interface {
 const (
 	// NetworkKeyPrefix is the prefix for network key in the kv store
 	NetworkKeyPrefix = "network"
+	// EndpointKeyPrefix is the prefix for endpoint key in the kv store
+	EndpointKeyPrefix = "endpoint"
 )
 
+var rootChain = []string{"docker", "libnetwork"}
+
 //Key provides convenient method to create a Key
 func Key(key ...string) string {
-	keychain := []string{"docker", "libnetwork"}
-	keychain = append(keychain, key...)
+	keychain := append(rootChain, key...)
 	str := strings.Join(keychain, "/")
 	return str + "/"
 }
 
+//UnKey provides convenient method to unpack the key to compliment the Key function
+func UnKey(key string) ([]string, error) {
+	chain := strings.Split(key, "/")
+	if len(chain) <= len(rootChain) || !reflect.DeepEqual(chain[0:len(rootChain)], rootChain) {
+		return nil, types.BadRequestErrorf("Invalid Key : %s", key)
+	}
+	return chain[len(rootChain):], nil
```

Comment:
When you split a key like this `/docker/libnetwork/a/b/` you would get the slice `chain`'s first element to be empty string and the last element to be empty string. So when you return `chain[len(rootChain):]` wouldn't it return the path element `libnetwork` and then the rest of the key and then lastly an empty string. This makes it distinctly asymmetric with the `Key` method above. Ideally when I do `Unkey(Key(<list_of_keys>))` we should be able to get the same list back. Not some thing else right?